src/HTSPTypes.h

```diff
@@ -151,17 +152,41 @@ struct SRecording
     priority  = 50; // kodi default - "normal"
   }
   
+  bool IsStopTimeInThePast () const
+  {
+    time_t now = time(NULL);
+    return (now > static_cast<time_t>(stop + (stopExtra * 60)));
+  }
+
   bool IsRecording () const
   {
-    return state == PVR_TIMER_STATE_COMPLETED ||
-           state == PVR_TIMER_STATE_ABORTED   ||
-           state == PVR_TIMER_STATE_RECORDING;
+    if (state != PVR_TIMER_STATE_ERROR)
+    {
+      // determine type by state.
+      return state == PVR_TIMER_STATE_COMPLETED ||
+             state == PVR_TIMER_STATE_ABORTED   ||
+             state == PVR_TIMER_STATE_RECORDING;
+    }
+    else
+    {
+      // determine type by time (both timers and recordings may have errors).
+      return IsStopTimeInThePast();
```

Comment:
The problem with showing items with an error in the recordings listing is - at least in Confluence theme - that there is no "status" row. One cannot see that a recording has an error...

But hey, then we should make things even more simple and display everything that has state PVR_TIMER_STATE_ERROR in the timer listing - withdraw the time-based decisions in IsTimer() and IsRecording()? Everything that has an error will be a timer - as per definition. What do you think?